docs/literate/src/files/adding_new_equations_literate/cubic_conservation_law_literate.jl

```diff
@@ -0,0 +1,236 @@
+# selected parameters, in this case the `solver`.
+
+## A new setup with dissipation
+semi = remake(semi, solver=DGSEM(3, flux_godunov))
+ode = semidiscretize(semi, tspan)
+sol = solve(ode, SSPRK43(),
+            save_everystep=false, callback=callbacks, maxiters=1e5);
+summary_callback()
+plot!(sol)
+#src # TODO: In the static tutorial version the plot (2) is slighly different than the executed one (2b).
+#src # Storing the link for possible changes.
+#src # ![tutorial_adding_new_equations_plot2](https://user-images.githubusercontent.com/12693098/111651740-c9196c80-8806-11eb-9a02-c0420eecf4fc.png)
+#src # ![tutorial_adding_new_equations_plot2b](https://user-images.githubusercontent.com/74359358/126664312-e2f2fc93-42b0-4082-a009-b4268b15ff21.png)
+
+# You can see that there are fewer oscillations, in particular around steep edges.
+# Now let's increase the final time (and also the spatial resolution).
+
+## A larger final time: Nonclassical shocks develop (you can even increase the refinement to 12)
+semi = remake(semi, mesh=TreeMesh(-1.0, 1.0, initial_refinement_level=8, n_cells_max=10^5))
+ode = semidiscretize(semi, (0.0, 0.5))
+sol = solve(ode, SSPRK43(),
+            save_everystep=false, callback=callbacks, maxiters=1e5);
+plot(sol)
+
+# You can observe that nonclassical shocks develop and are stable under grid refinement,
+# e.g. for `initial_refinement_level=12`. In this case, these nonclassical shocks
+# can be avoided by using an entropy-dissipative semidiscretization. Thus, we need
+# to define an entropy-conservative numerical flux
+
+@inline function Trixi.flux_ec(u_ll, u_rr, orientation, equation::CubicConservationLaw.CubicEquation)
+  return SVector(0.25 * (u_ll[1]^3 + u_ll[1]^2 * u_rr[1] + u_ll[1] * u_rr[1]^2 + u_rr[1]^3))
+end
+
+# and use a [`VolumeIntegralFluxDifferencing`](@ref) instead of the standard
+# [`VolumeIntegralWeakForm`](@ref) in the DGSEM.
+
+## Let's use a provably entropy-dissipative semidiscretization
+semi = remake(semi, solver=DGSEM(3, flux_godunov, VolumeIntegralFluxDifferencing(flux_ec)))
+ode = semidiscretize(semi, (0.0, 0.5))
+sol = solve(ode, SSPRK43(),
+            save_everystep=false, callback=callbacks, maxiters=1e5);
+plot(sol)
+
+# Possible next steps could be
+# - to define `Trixi.max_abs_speeds(u, equations::CubicEquation) = 3 * u[1]^2`
+#   to use CFL-based time step control via a [`StepsizeCallback`](@ref)
+# - to define quantities of interest like `Trixi.entropy(u, equations::CubicEquation) = u[1]^2`
+#   and integrate them in a simulation using the [`AnalysisCallback`](@ref)
+# - to experiment with shock-capturing volume integrals [`VolumeIntegralShockCapturingHG`](@ref)
+#   and adaptive mesh refinement [`AMRCallback`](@ref)
+
+
+# ## Summary of the code
+
+# To sum up, here is the complete code that we used (without the [`SummaryCallback`](@ref)
+# since that creates a lot of unnecessary output in the doctests of this tutorial).
+
+## Define new physics
+module CubicConservationLaw
+
```

Comment:
Redefining the module leads to the `Warning` in the end of the code summary. I don't see any easy way to fix this right now.